src/server/v2.0/handler/project.go

```diff
@@ -63,3 +241,380 @@ func (a *projectAPI) GetLogs(ctx context.Context, params operation.GetLogsParams
+		query.Keywords["owner"] = owner
+	}
+	if params.Public != nil {
+		query.Keywords["public"] = lib.BoolValue(params.Public)
+	}
+
+	secCtx, ok := security.FromContext(ctx)
+	if ok && secCtx.IsAuthenticated() {
+		if !secCtx.IsSysAdmin() && !secCtx.IsSolutionUser() {
+			// authenticated but not system admin or solution user,
+			// return public projects and projects that the user is member of
+			if l, ok := secCtx.(*local.SecurityContext); ok {
+				currentUser := l.User()
+				member := &project.MemberQuery{
+					Name:     currentUser.Username,
+					GroupIDs: currentUser.GroupIDs,
+				}
+
+				// not filter by public or filter by the public with true,
+				// so also return public projects for the member
+				if public, ok := query.Keywords["public"]; !ok || lib.ToBool(public) {
+					member.WithPublic = true
+				}
+
+				query.Keywords["member"] = member
+			} else {
+				// can't get the user info, force to return public projects
+				query.Keywords["public"] = true
+			}
+		}
+	} else {
+		if params.Public != nil && !*params.Public {
+			// anonymous want to query private projects return empty projects directly
+			return operation.NewListProjectsOK().WithXTotalCount(0).WithPayload([]*models.Project{})
+		}
+		// force to return public projects for anonymous
+		query.Keywords["public"] = true
+	}
+
+	total, err := a.projectCtl.Count(ctx, query)
+	if err != nil {
+		return a.SendError(ctx, err)
+	}
+
+	projects, err := a.projectCtl.List(ctx, query, project.WithCVEAllowlist(), project.WithOwner())
+	if err != nil {
+		return a.SendError(ctx, err)
+	}
+
+	var wg sync.WaitGroup
+	for _, p := range projects {
+		wg.Add(1)
+		go func(p *project.Project) {
+			defer wg.Done()
+			// due to the issue https://github.com/lib/pq/issues/81 of lib/pg or postgres,
+			// simultaneous queries in transaction may failed, so clone a ctx with new ormer here
+			if err := a.populateProperties(orm.Clone(ctx), p); err != nil {
+				log.G(ctx).Errorf("failed to populate propertites for project %s, error: %v", p.Name, err)
+			}
+		}(p)
+	}
+	wg.Wait()
+
+	var payload []*models.Project
+	for _, p := range projects {
+		payload = append(payload, model.NewProject(p).ToSwagger())
+	}
+
+	return operation.NewListProjectsOK().
+		WithXTotalCount(total).
+		WithLink(a.Links(ctx, params.HTTPRequest.URL, total, query.PageNumber, query.PageSize).String()).
+		WithPayload(payload)
+}
+
+func (a *projectAPI) UpdateProject(ctx context.Context, params operation.UpdateProjectParams) middleware.Responder {
+	if err := a.RequireProjectAccess(ctx, params.ProjectID, rbac.ActionUpdate); err != nil {
+		return a.SendError(ctx, err)
+	}
+
+	p, err := a.projectCtl.Get(ctx, params.ProjectID, project.Metadata(false))
+	if err != nil {
+		return a.SendError(ctx, err)
+	}
+
+	lib.JSONCopy(&p.Metadata, params.Project.Metadata)
+
+	if params.Project.CVEAllowlist != nil {
+		// force the project id of the cve allowlist to be the project which will be update
```

Comment:
give more clear comments? I think if the PID in the parameters is not equals with the ID in path, just give a 400 or 403？